### ISSUE TITLE
[Feature]: vectorise the bucketed decode metadata precompute

### ISSUE THREAD

**Issue by jvlunteren** [torch-spyre/spyre-inference#779]
### Feature description

Vectorise the two element-at-a-time fills in `SpyreAttentionMetadataBuilder.build()`'s bucketed decode precompute, so the per-step host cost stops scaling with `num_seqs * num_active`.

Both currently write one element or one row at a time:

```
block_ids_padded_cpu = torch.zeros(b_blocks * b_seqs, dtype=torch.int32)
for s, n in enumerate(num_active):
    for b in range(min(n, b_blocks)):
        block_ids_padded_cpu[b * b_seqs + s] = block_table[s, b]   # scalar store
...
for s in range(num_seqs):
    for b in range(min(num_active[s], b_blocks)):
        mask_bs_bb[s, b] = attention_mask_tiles[s][b][0]            # row copy
```

### Motivation and context

`build()` runs on every decode step, and vLLM calls it from `_prepare_inputs` before the model forward rather than alongside it (gpu_model_runner.py:2607 against the `forward` at :3981), so this is host time on the critical path rather than work hidden behind device execution. The Spyre backend also does not implement supports_update_block_table, so there is no cached-metadata path that skips the rebuild.

The cost of both loops grows with `num_seqs × num_active`, which means it grows with batch width and with context length. That makes it a mixed-batch and long-context problem specifically: a wide batch of long sequences pays the most, and those are the cases where the bucketed path is otherwise most attractive.

Worth noting the same class of fix was applied elsewhere recently in #728 (`vectorize-vocab-embedding-table`s), so the pattern is established in this codebase.

### Proposed solution

Fill the block ids row-major with a single gather and transpose, and build each sequence's mask rows with one stack rather than a copy per block.

Both replacements must be exact, not approximate. The padded entries and the `-inf` positions carry meaning downstream. In particular the block-id fill has to stay correct for ragged `num_active` (sequences at different context lengths), for a sequence whose block count exceeds the bucket (truncation), and for rows past the batch, which the kernel relies on being zero. A test comparing the vectorised fill against an explicit per-element fill across those shapes is the cheapest way to keep it honest, and needs no device.

### Checklist

- [x] I have searched for similar feature requests

**Comment by tdoublep** [comment]
Done.
